src/v3/src/transformer/uischema/setLtrFields.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  FieldElement,
+  TransformStepFn,
+} from '../../types';
+import { traverseLayout } from '../util';
+
+export const setLtrFields: TransformStepFn = (formbag) => {
+  traverseLayout({
+    layout: formbag.uischema,
+    predicate: (el) => {
+      if (el.type === 'Field') {
+        // fields with these names should always be LTR
+        const ltrOnlyFieldNames = ['credentials.passcode', 'identifier', 'credentials.newPassword', 'confirmPassword'];
+        const fieldElement = (el as FieldElement);
+        if (ltrOnlyFieldNames.includes(fieldElement.options.inputMeta.name)) {
+          return true;
+        }
+      }
+      return false;
+    },
+    callback: (el) => {
+      const fieldElement = (el as FieldElement);
```

Comment:
I know this is 'safe' because we are counting on the predicate to only run this callback if it was `type === 'Field'` but probably best to still run a check inside this function in case we ever change the logic above. Ideally we could rework this transformer function to have some kind generic with type-narrowing to guarantee the incoming type but I'm not clever enough to think of how to do that right now :P

Reply:
haha yeah that does make sense to check in the callback aswell, will add that in 👍